client/src/components/chats/FriendsStartAChat.js

```diff
@@ -0,0 +1,19 @@
+import FriendStartAChat from './FriendStartAChat'
+
+function FriendsStartAChat({ friends }) {
+  return (
+    <>
```

Comment:
May I suggest that this component returns a form method="dialog". The children can be input elements with  type="submit". This way the dialog is closed upon clicking a friend.

You may set the currentFriendId like so:
import { setMessages } from '../../store/features/messages/messagesSlice'
const dispatch = useDispatch()
const handleClick = () => {
  dispatch(setCurrentFriendId(currentFriendId))
}

Reply:
Thanks, I'll do that.

Reply:
So in our click handler, after setting the currentFriendId we can add the following:
```
window.location.replace(
   `${ process.env.REACT_APP_FRONTEND_SERVER_URL }/chatroom`
)
```

Reply:
Wow, I would never think of that. I tried working on "window" before and I was getting errors. But I'll try it. Thanks

Reply:
You're welcome. I learned this while working on my solo project when I had to redirect after login/signup.

Reply:
The window function didn't work for me. Perhaps there is a React way to solve redirecting, I'll do some research.

Reply:
How about using `{Link}` from `react-router-dom`?
```
<Link to="/chatroom">
   latest message
</Link>
```